src/databricks/labs/ucx/hive_metastore/tables.py

```diff
@@ -252,3 +285,67 @@ def _init_seen_tables(self):
 
     def _table_already_upgraded(self, target) -> bool:
         return target in self._seen_tables
+
+    def revert_migrated_tables(
+        self, schema: str | None = None, table: str | None = None, *, deletemanaged: bool = False
+    ):
+        upgraded_tables = []
+        if table and not schema:
+            logger.error("Cannot accept 'Table' parameter without 'Schema' parameter")
+        for cur_table in self._tc.snapshot():
+            if schema and cur_table.database != schema:
+                continue
+            if table and cur_table.name != table:
+                continue
+            if cur_table.upgraded_to is not None:
+                upgraded_tables.append(cur_table)
+
+        for upgraded_table in upgraded_tables:
+            if upgraded_table.object_type == "EXTERNAL" or deletemanaged:
+                logger.info(
+                    f"Reverting {upgraded_table.object_type} Table {upgraded_table.database}.{upgraded_table.name} "
+                    f"upgraded_to {upgraded_table.upgraded_to}"
+                )
+                self._backend.execute(upgraded_table.sql_unset_to("hive_metastore"))
+                self._backend.execute(upgraded_table.sql_remove_target())
+            else:
+                logger.info(
+                    f"Skipping {upgraded_table.object_type} Table {upgraded_table.database}.{upgraded_table.name} "
+                    f"upgraded_to {upgraded_table.upgraded_to}"
+                )
+
+        self._tc.unset_upgraded_to(schema, table, deletemanaged=deletemanaged)
+
+    def get_migrated_count(self, schema: str | None = None, table: str | None = None) -> list[MigrationCount]:
+        migration_list = []
+        table_by_database = defaultdict(list)
+        for cur_table in self._tc.snapshot():
+            if schema and cur_table.database != schema:
+                continue
+            if table and cur_table.name != table:
+                continue
+            if cur_table.upgraded_to is not None:
+                table_by_database[cur_table.database].append(cur_table)
+
+        for cur_database in table_by_database.keys():
+            external_tables = 0
+            managed_tables = 0
+            for current_table in table_by_database[cur_database]:
+                if current_table.upgraded_to is not None:
+                    if current_table.object_type == "EXTERNAL":
+                        external_tables += 1
+                if current_table.object_type == "MANAGED":
+                    managed_tables += 1
+            migration_list.append(
+                MigrationCount(database=cur_database, managed_tables=managed_tables, external_tables=external_tables)
+            )
+        return migration_list
+
+    def checks_upgraded_to(self, schema: str, table: str):
```

Comment:
```suggestion
    def is_upgraded(self, schema: str, table: str) -> bool:
```
This is a better name for this method

Reply:
OK